engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -902,14 +902,14 @@ private void deployNodes(List<OperatorDeployInfo> nodeList) throws IOException
 
       Context parentContext;
       if (ndi instanceof UnifierDeployInfo) {
-        OperatorContext unifiedOperatorContext = new OperatorContext(0, ((UnifierDeployInfo)ndi).operatorAttributes, containerContext);
+        OperatorContext unifiedOperatorContext = new OperatorContext(ndi, containerContext);
         parentContext = new PortContext(ndi.inputs.get(0).contextAttributes, unifiedOperatorContext);
         massageUnifierDeployInfo(ndi);
       } else {
         parentContext = containerContext;
       }
 
-      OperatorContext ctx = new OperatorContext(ndi.id, ndi.contextAttributes, parentContext);
+      OperatorContext ctx = new OperatorContext(ndi, parentContext);
```

Comment:
Please check the old code behavior. Unless I miss something, old code will use both operatorAttributes and contextAttributes with UnifierDeployInfo.